blocks/api/registration.js

```diff
@@ -80,10 +80,14 @@ export function registerBlockType( name, settings ) {
 		return;
 	}
 	if ( blocks[ name ] ) {
-		console.error(
-			'Block "' + name + '" is already registered.'
-		);
-		return;
+		if ( module.hot ) {
```

Comment:
I'm not a huge fan of any code aside from the entry point file being HMR-aware, but I suppose this is a consequence of us having implemented the block registry as a shared singleton?